File: app/src/main/java/com/anokmik/tripassistant/trip/list/TripsListContract.java
package com.anokmik.tripassistant.trip.list;

import android.support.annotation.LayoutRes;

import com.anokmik.persistence.model.Trip;
import com.anokmik.tripassistant.databinding.adapter.ViewHolderPresenter;

import java.util.List;

public interface TripsListContract {

    interface View {

        @LayoutRes
        int getRowItemLayoutId();

        int getItemBindingId();

        void showTrip(Trip trip);

    }

    interface Presenter {

        List<Trip> getTrips();

        ViewHolderPresenter<Trip> getViewHolderPresenter();

    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/trip/list/TripsListContract.kt
package com.anokmik.tripassistant.trip.list

import com.anokmik.persistence.model.Trip
import com.anokmik.tripassistant.databinding.adapter.ViewHolderPresenter

interface TripsListContract {

    interface View {

        val rowItemLayoutId: Int

        val itemBindingId: Int

        fun showTrip(trip: Trip?)

    }

    interface Presenter {

        val trips: List<Trip?>

        val viewHolderPresenter: ViewHolderPresenter<Trip>

    }

}